cluster-api/src/main/java/io/aiven/klaw/clusterapi/services/MonitoringService.java

```diff
@@ -0,0 +1,94 @@
+  public List<HashMap<String, String>> getConsumerGroupDetails(
+      String consumerGroupId,
+      String topicName,
+      String environment,
+      KafkaSupportedProtocol protocol,
+      String clusterName)
+      throws Exception {
+    log.info(
+        "getConsumerGroupDetails topicName {} consumerGroupId {} environment {} protocol {} clusterName {}",
+        topicName,
+        consumerGroupId,
+        environment,
+        protocol,
+        clusterName);
+
+    AdminClient adminClient = clusterApiUtils.getAdminClient(environment, protocol, clusterName);
+
+    List<HashMap<String, String>> consumerGroupOffsetList = new ArrayList<>();
+    HashMap<String, String> offsetDetails;
+    try {
+      DescribeTopicsResult describeTopicsResult =
+          adminClient.describeTopics(Collections.singletonList(topicName));
+      List<TopicPartitionInfo> topicPartitions =
+          describeTopicsResult.values().get(topicName).get().partitions();
+
+      TopicPartition topicPartition;
+      Map<TopicPartition, OffsetSpec> topicPartitionOffsetSpecMap = new HashMap<>();
+      topicPartitionOffsetSpecMap = new HashMap<>();
+
+      for (TopicPartitionInfo topicPartitionInfo : topicPartitions) {
+        topicPartition = new TopicPartition(topicName, topicPartitionInfo.partition());
+        topicPartitionOffsetSpecMap.put(topicPartition, OffsetSpec.earliest());
+      }
+      ListOffsetsResult listOffsetsEarliestResult =
+          adminClient.listOffsets(topicPartitionOffsetSpecMap);
+
+      topicPartitionOffsetSpecMap = new HashMap<>();
+      for (TopicPartitionInfo topicPartitionInfo : topicPartitions) {
+        topicPartition = new TopicPartition(topicName, topicPartitionInfo.partition());
+        topicPartitionOffsetSpecMap.put(topicPartition, OffsetSpec.latest());
+      }
+
+      ListOffsetsResult listOffsetsLatestResult =
+          adminClient.listOffsets(topicPartitionOffsetSpecMap);
+
+      for (TopicPartitionInfo topicPartitionInfo : topicPartitions) {
+        topicPartition = new TopicPartition(topicName, topicPartitionInfo.partition());
+
+        offsetDetails = new HashMap<>();
+        long earliestOffset =
+            listOffsetsEarliestResult.partitionResult(topicPartition).get().offset();
+        long latestOffset = listOffsetsLatestResult.partitionResult(topicPartition).get().offset();
+        long lag = latestOffset - earliestOffset;
+
+        offsetDetails.put("topicPartitionId", topicPartition.partition() + "");
+        offsetDetails.put("currentOffset", earliestOffset + "");
+        offsetDetails.put("endOffset", latestOffset + "");
+        offsetDetails.put("lag", lag + "");
+        consumerGroupOffsetList.add(offsetDetails);
+      }
+      return consumerGroupOffsetList;
+    } catch (Exception exception) {
+      log.error(
+          "Cannot retrieve consumer offset details topicName: {} groupid: {} Error: {}",
+          topicName,
+          consumerGroupId,
+          exception.toString());
```

Comment:
```suggestion
      log.error(
          "Cannot retrieve consumer offset details topicName: {} groupid: {} Error: ",
          topicName,
          consumerGroupId,
          exception);
```
it's better to log the whole trace